### ISSUE TITLE
ConvolutionLayer pretrain does not work 

### ISSUE THREAD

**Issue by nyghtowl** [deeplearning4j/deeplearning4j#656]
pretrain calls optimizer, which calls computeGradientAndScore. CNN is using computeGradientAndScore from BaseLayer which works for backprop but in pretrain, gradient is not set and is null. A nullPointException is thrown. Need a variation in CNN to calculate gradient for just pretrain.

**Comment by agibsonccc** [comment]
We would just want to skip it here. There is no "pretrain" or "fit" The
convolutional layers would only work for backprop.
ᐧ

**Comment by nyghtowl** [comment]
Addressed in #660
